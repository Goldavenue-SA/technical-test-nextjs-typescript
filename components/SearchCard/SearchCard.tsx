import { Card } from "../Card/Card";
import { Counter } from "../Counter/Counter";
import { SearchInput } from "../SearchInput/SearchInput";
import { SearchInputContainer } from "./Styles";

export const SearchCard = () => {
  return (
    <>
      <Card>
        <SearchInputContainer>
          <SearchInput
            label="Search"
            placeholder="Search a Pokemon Name"
            onSearch={() => ""}
            id="pokemon-name"
          />

          <SearchInput
            label="Power threshold"
            placeholder="Write a Power Threshold"
            onSearch={() => ""}
            id="pokemon-power-treshold"
          />
        </SearchInputContainer>
        <Counter countOverThreshold={329} min={328} max={456} />
      </Card>
    </>
  );
};
