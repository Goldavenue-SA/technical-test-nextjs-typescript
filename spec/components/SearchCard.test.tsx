import { render } from '@testing-library/react';
import type { ReactElement } from 'react';
import { SearchCard } from '../../components/SearchCard/SearchCard';

describe('Test Search Card Component', () => {
  it('should match snapshot with a mock page', () => {
    expect(render(<SearchCard />)).toMatchSnapshot();
  });
});
