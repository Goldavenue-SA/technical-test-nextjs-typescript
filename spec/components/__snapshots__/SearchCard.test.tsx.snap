// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Card Component should match snapshot with a mock page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-5477665f-0 eHVTwR"
      >
        <div
          class="sc-67d6c285-0 ddkiWA"
        >
          <div
            class="sc-3134fa0a-0 laGSnx"
          >
            <label
              class="sc-3134fa0a-2 hMoKSw"
              for="pokemon-name"
            >
              Search
            </label>
            <input
              aria-describedby="search-help-text"
              aria-label="Search input"
              class="sc-3134fa0a-1 eZYfBK"
              id="pokemon-name"
              placeholder="Search a Pokemon Name"
              type="search"
            />
          </div>
          <div
            class="sc-3134fa0a-0 laGSnx"
          >
            <label
              class="sc-3134fa0a-2 hMoKSw"
              for="pokemon-power-treshold"
            >
              Power threshold
            </label>
            <input
              aria-describedby="search-help-text"
              aria-label="Search input"
              class="sc-3134fa0a-1 eZYfBK"
              id="pokemon-power-treshold"
              placeholder="Write a Power Threshold"
              type="search"
            />
          </div>
        </div>
        <section
          class="sc-5d04bb7b-0 iLyYNB"
        >
          <p
            aria-live="polite"
            role="status"
          >
            Count over threshold:
            <span>
              <strong>
                329
              </strong>
            </span>
          </p>
          <p>
            Min:
            <span>
              <strong>
                328
              </strong>
            </span>
          </p>
          <p>
            Max:
            <span>
              <strong>
                456
              </strong>
            </span>
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-5477665f-0 eHVTwR"
    >
      <div
        class="sc-67d6c285-0 ddkiWA"
      >
        <div
          class="sc-3134fa0a-0 laGSnx"
        >
          <label
            class="sc-3134fa0a-2 hMoKSw"
            for="pokemon-name"
          >
            Search
          </label>
          <input
            aria-describedby="search-help-text"
            aria-label="Search input"
            class="sc-3134fa0a-1 eZYfBK"
            id="pokemon-name"
            placeholder="Search a Pokemon Name"
            type="search"
          />
        </div>
        <div
          class="sc-3134fa0a-0 laGSnx"
        >
          <label
            class="sc-3134fa0a-2 hMoKSw"
            for="pokemon-power-treshold"
          >
            Power threshold
          </label>
          <input
            aria-describedby="search-help-text"
            aria-label="Search input"
            class="sc-3134fa0a-1 eZYfBK"
            id="pokemon-power-treshold"
            placeholder="Write a Power Threshold"
            type="search"
          />
        </div>
      </div>
      <section
        class="sc-5d04bb7b-0 iLyYNB"
      >
        <p
          aria-live="polite"
          role="status"
        >
          Count over threshold:
          <span>
            <strong>
              329
            </strong>
          </span>
        </p>
        <p>
          Min:
          <span>
            <strong>
              328
            </strong>
          </span>
        </p>
        <p>
          Max:
          <span>
            <strong>
              456
            </strong>
          </span>
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
